MetaMask/metamask-extension
# Caip multichain caip 27 api spec tests

## **Description**
This adds api spec tests that utilize `externally_connectable` as well as `caip-27` requests to wrap the "legacy" api-spec tests.

It refactors a lot into `transform.ts` to help reduce duplication since we want to test it the same way as our current api-spec tests.



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27229?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. `CHAIN_PERMISSIONS=1 BARAD_DUR=1 yarn build:test`
2. `yarn test:api-spec-multichain`
3. see multiple html reporters pop up with all passing.

## **Screenshots/Recordings**
<img width="1309" alt="image" src="https://github.com/user-attachments/assets/035a1725-b215-4d7c-982c-9867814113b4">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
